main.go

```diff
@@ -145,40 +111,62 @@ func main() {
 
 		webhookOpts.CARotateInterval, err = environment.LookupAsDuration("CA_ROTATE_INTERVAL")
 		if err != nil {
-			log.Error(err, "Failed retrieving ca rotate interval")
+			setupLog.Error(err, "Failed retrieving ca rotate interval")
 			os.Exit(1)
 		}
 
 		webhookOpts.CAOverlapInterval, err = environment.LookupAsDuration("CA_OVERLAP_INTERVAL")
 		if err != nil {
-			log.Error(err, "Failed retrieving ca overlap interval")
+			setupLog.Error(err, "Failed retrieving ca overlap interval")
 			os.Exit(1)
 		}
 
 		webhookOpts.CertRotateInterval, err = environment.LookupAsDuration("CERT_ROTATE_INTERVAL")
 		if err != nil {
-			log.Error(err, "Failed retrieving cert rotate interval")
+			setupLog.Error(err, "Failed retrieving cert rotate interval")
 			os.Exit(1)
 		}
 
 		if err := webhook.AddToManager(mgr, webhookOpts); err != nil {
-			log.Error(err, "Cannot initialize webhook")
+			setupLog.Error(err, "Cannot initialize webhook")
+			os.Exit(1)
+		}
+	} else if environment.IsOperator() {
+		// +kubebuilder:scaffold:builder
+		if err = (&controllers.NMStateReconciler{
+			Client: mgr.GetClient(),
+			Log:    ctrl.Log.WithName("controllers").WithName("NMState"),
+			Scheme: mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create NMState controller", "controller", "NMState")
+			os.Exit(1)
+		}
+	} else if environment.IsHandler() {
+		// +kubebuilder:scaffold:builder
+		if err = (&controllers.NodeReconciler{
+			Client: mgr.GetClient(),
+			Log:    ctrl.Log.WithName("controllers").WithName("Node"),
+			Scheme: mgr.GetScheme(),
+		}).SetupWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create Node controller", "controller", "NMState")
 			os.Exit(1)
 		}
-	} else {
-		// Setup all Controllers
-		if err := controller.AddToManager(mgr); err != nil {
-			log.Error(err, "Cannot initialize controller")
+		// +kubebuilder:scaffold:builder
```

Comment:
Is this really needed twice? Isn't the scaffold only inserting code?

Reply:
I am going to remove those, since we have a very custom main migration has to be done manually, at least let's keep code similar to emitted one.